src/aus.h

```diff
@@ -0,0 +1,329 @@
+            #ifdef _WIN32
+            return "Windows";
+            #else
+            #ifdef __APPLE__
+            return "Mac OS";
+            #else
+            #ifdef __linux__
+            return "Linux";
+            #else
+            return "Unknown";
+            #endif
+            #endif
+            #endif
+        }
+
+        
+        void set(std::string key, long value)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->set(value);
+            }
+            else
+            {//_mp[key] = std::make_unique<aus_counter>(new aus_counter(value));
+                _mp[key] = new aus_counter(value);
+            }
+
+        }
+
+        void increment(std::string key)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->increment();
+            }
+            else
+            {
+               // _mp[key] = std::make_unique<aus_counter>(new aus_counter(1));
+                _mp[key] = new aus_counter(1);
+            }
+        }
+
+        void decrement(std::string key)
+        {
+            if (_mp.find( key ) != _mp.end() && _mp[key] >0)
+            {
+                _mp[key]->decrement();
+            }
+        }
+
+        long get(std::string key)
+        {
+            assert_key_exists(key);
+            return _mp[key]->get();
+        }
+
+        void start(std::string key)
+        {
+            if (_mp.find(key) == _mp.end())
+            {
+                _mp.insert(std::make_pair(key, new aus_timer()));
+                _mp[key]->start();
+            }
+            else
+            {
+                _mp[key]->start();
+            }
+        }
+
+        void stop(std::string key)
+        {
+            assert_key_exists(key);
+            _mp[key]->stop();
+        }
+
+        std::vector<std::string> get_counters_list()
+        {
+            std::vector<std::string> result;
+            for (const auto& pair : _mp) {
+                result.push_back(pair.first);
+            }
+            return result;
+        }
+
+        bool device_exist( std::string serial )
+        {
+            if ( _mp_devices_manager.find( serial ) == _mp_devices_manager.end())
+            {
+                return false;
+            }
+            return true;
+        }
+
+        void insert_device_to_device_manager( std::string serial, std::string name )
+        {
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES" ));
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES", name ) );
+            _mp_devices_manager[serial]=name;
+        }
+
+        void on_device_changed(std::string serial, std::string name )
+        {
+            if ( !device_exist( serial ) )
+            {
+                insert_device_to_device_manager( serial, name );
+            }
+        } 
+
+    private:
+        std::unordered_map<std::string, aus_value*> _mp;
```

Comment:
Let's add a comment about what is the map key / value stands for.
Also what `_mp` means?